### ISSUE TITLE
firebase_analytics 7.0.1, cannot build on iOS, CocoaPods could not find compatible versions for pod "firebase_analytics

### ISSUE THREAD

**Issue by luix93** [FirebaseExtended/flutterfire#4709]
## Bug report

**Describe the bug**
After implementing Firebase Analytics i get the following error while trying to build:

```
   [!] CocoaPods could not find compatible versions for pod "firebase_analytics":
      In Podfile:
        firebase_analytics (from `.symlinks/plugins/firebase_analytics/ios`)

    Specs satisfying the `firebase_analytics (from `.symlinks/plugins/firebase_analytics/ios`)` dependency were found, but they required a higher minimum
    deployment target.

```
### Steps to reproduce

Added to my pubspec.yaml file:

`firebase_analytics: ^7.0.1`

**Comment by cwyu-Ivan** [comment]
I am having the same problem on 7.0.0, I tried deleting PodFile and PodFile.lock, flutter clean, pod repo update but those methods don't work.

**Comment by badaween-ahmed** [comment]
I am getting the same result.
```

Analyzing dependencies
Pre-downloading: `FirebaseFirestore` from `https://github.com/invertase/firestore-ios-sdk-frameworks.git`, tag `7.3.0`
cloud_firestore: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
cloud_functions: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
firebase_auth: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
firebase_core: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
firebase_remote_config: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
firebase_storage: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
[!] CocoaPods could not find compatible versions for pod "cloud_functions":
  In Podfile:
    cloud_functions (from `.symlinks/plugins/cloud_functions/ios`)

Specs satisfying the `cloud_functions (from `.symlinks/plugins/cloud_functions/ios`)` dependency were found, but they required a higher minimum deployment target.
```

I even tried with different versions which mentioned here https://github.com/invertase/firestore-ios-sdk-frameworks
I tried will all sorts of commands related to this issue from StackOverflow or GitHub issue.

**Comment by badaween-ahmed** [comment]
`flutter run`

```
Launching lib/main.dart on iPhone 12 Pro in debug mode...
Running pod install...                                              2.5s
CocoaPods' output:
↳
      Preparing

    Analyzing dependencies

    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)

    Fetching external sources
    -> Pre-downloading: `FirebaseFirestore` from `https://github.com/invertase/firestore-ios-sdk-frameworks.git`, tag `7.3.0`
      > Copying FirebaseFirestore from `/Users/ahmed/Library/Caches/CocoaPods/Pods/External/FirebaseFirestore/16c2cc8f520a5e8df4ee73a51d0a1966` to
      `Pods/FirebaseFirestore`
    -> Fetching podspec for `Flutter` from `Flutter`
    -> Fetching podspec for `animated_theme_switcher` from `.symlinks/plugins/animated_theme_switcher/ios`
    -> Fetching podspec for `cloud_firestore` from `.symlinks/plugins/cloud_firestore/ios`
    cloud_firestore: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
    -> Fetching podspec for `cloud_functions` from `.symlinks/plugins/cloud_functions/ios`
    cloud_functions: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
    -> Fetching podspec for `device_info` from `.symlinks/plugins/device_info/ios`
    -> Fetching podspec for `firebase_auth` from `.symlinks/plugins/firebase_auth/ios`
    firebase_auth: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
    -> Fetching podspec for `firebase_core` from `.symlinks/plugins/firebase_core/ios`
    firebase_core: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
    -> Fetching podspec for `firebase_remote_config` from `.symlinks/plugins/firebase_remote_config/ios`
    firebase_remote_config: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
    -> Fetching podspec for `firebase_storage` from `.symlinks/plugins/firebase_storage/ios`
    firebase_storage: Using Firebase SDK version '7.3.0' defined in 'firebase_core'
    -> Fetching podspec for `flutter_keyboard_visibility` from `.symlinks/plugins/flutter_keyboard_visibility/ios`
    -> Fetching podspec for `flutter_local_notifications` from `.symlinks/plugins/flutter_local_notifications/ios`
    -> Fetching podspec for `flutter_native_timezone` from `.symlinks/plugins/flutter_native_timezone/ios`
    -> Fetching podspec for `google_sign_in` from `.symlinks/plugins/google_sign_in/ios`
    -> Fetching podspec for `in_app_purchase` from `.symlinks/plugins/in_app_purchase/ios`
    -> Fetching podspec for `package_info` from `.symlinks/plugins/package_info/ios`
    -> Fetching podspec for `path_provider` from `.symlinks/plugins/path_provider/ios`
    -> Fetching podspec for `shared_preferences` from `.symlinks/plugins/shared_preferences/ios`
    -> Fetching podspec for `url_launcher` from `.symlinks/plugins/url_launcher/ios`

    Resolving dependencies of `Podfile`
      CDN: trunk Relative path: CocoaPods-version.yml exists! Returning local because checking is only perfomed in repo update
    [!] CocoaPods could not find compatible versions for pod "cloud_functions":
      In Podfile:
        cloud_functions (from `.symlinks/plugins/cloud_functions/ios`)

    Specs satisfying the `cloud_functions (from `.symlinks/plugins/cloud_functions/ios`)` dependency were found, but they required a higher minimum
    deployment target.

    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:328:in `raise_error_unless_state'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:310:in `block in unwind_for_conflict'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:308:in `tap'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:308:in `unwind_for_conflict'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:684:in `attempt_to_activate'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:254:in `process_topmost_state'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:182:in `resolve'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.6.6/lib/molinillo/resolver.rb:43:in `resolve'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/resolver.rb:94:in `resolve'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer/analyzer.rb:1074:in `block in resolve_dependencies'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/user_interface.rb:64:in `section'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer/analyzer.rb:1072:in `resolve_dependencies'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer/analyzer.rb:124:in `analyze'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:414:in `analyze'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:239:in `block in resolve_dependencies'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/user_interface.rb:64:in `section'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:238:in `resolve_dependencies'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/installer.rb:160:in `install!'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/command/install.rb:52:in `run'
    /Library/Ruby/Gems/2.6.0/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/lib/cocoapods/command.rb:52:in `run'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.1/bin/pod:55:in `<top (required)>'
    /usr/local/bin/pod:23:in `load'
    /usr/local/bin/pod:23:in `<main>'

Error running pod install
Error launching application on iPhone 12 Pro.
```

**Comment by ercadev** [comment]
As mentioned in the [changelog](https://pub.dev/packages/firebase_analytics/changelog#700) the minimum deployment target is now iOS 10. 

Go to your Podfile and uncomment
`# platform :ios, '9.0'`

Then change the version to 10
`platform :ios, '10.0'`

**Comment by badaween-ahmed** [comment]
ohh yeah I changed to `platform :ios, '10.0'`and then, it worked I didn't say here becuase I didn't know why it worked lol. Thanks for the explaination.

**Comment by markusaksli-nc** [comment]
As mentioned by @ercadev this is was the whole reason for the recent version bumps in all of the FlutterFire plugins.

Closing since this is mentioned in both the changelogs and https://firebase.flutter.dev/docs/installation/ios#ios-supported-versions

**Comment by njustin** [comment]
> As mentioned by @ercadev this is was the whole reason for the recent version bumps in all of the FlutterFire plugins.
> 
> Closing since this is mentioned in the error message itself, the changelogs, and https://firebase.flutter.dev/docs/installation/ios#ios-supported-versions

Please, do not forget to also update the Migration Guide with this instruction.

**Comment by nilsreichardt** [comment]
> As mentioned in the [changelog](https://pub.dev/packages/firebase_analytics/changelog#700) the minimum deployment target is now iOS 10.
> 
> Go to your Podfile and uncomment
> `# platform :ios, '9.0'`
> 
> Then change the version to 10
> `platform :ios, '10.0'`

I have to use `platform :ios, 11.0` and this doesn't fix the issue. `platform :ios, 10.0` resolves this issue, but an other package (`jitsi_meet`) requires `platform :ios, 11.0` so I can't set it to `10.0`. Any other solutions?

**Comment by nilsreichardt** [comment]
> > As mentioned in the [changelog](https://pub.dev/packages/firebase_analytics/changelog#700) the minimum deployment target is now iOS 10.
> > Go to your Podfile and uncomment
> > `# platform :ios, '9.0'`
> > Then change the version to 10
> > `platform :ios, '10.0'`
> 
> I have to use `platform :ios, 11.0` and this doesn't fix the issue. `platform :ios, 10.0` resolves this issue, but an other package (`jitsi_meet`) requires `platform :ios, 11.0` so I can't set it to `10.0`. Any other solutions?
> 
> ```
> [!] CocoaPods could not find compatible versions for pod "Firebase/CoreOnly":
>   In Podfile:
>     firebase_analytics (from `.symlinks/plugins/firebase_analytics/ios`) was resolved to 0.0.1, which depends on
>       Firebase/Analytics (~> 6.0) was resolved to 6.0.0, which depends on
>         Firebase/Core (= 6.0.0) was resolved to 6.0.0, which depends on
>           Firebase/CoreOnly (= 6.0.0)
> 
>     firebase_core (from `.symlinks/plugins/firebase_core/ios`) was resolved to 0.7.0, which depends on
>       Firebase/CoreOnly (= 7.4.0)
> ```

1. `flutter clean`
2. `pod update`

Fixed it 💙
